indico/modules/events/registration/controllers/management/regforms.py

```diff
@@ -214,24 +214,37 @@ class RHRegistrationFormOpen(RHManageRegFormBase):
     """Opens registration for a registration form"""
 
     def _process(self):
+        old_dts = (self.regform.start_dt, self.regform.end_dt)
         if self.regform.has_ended:
             self.regform.end_dt = None
         else:
             self.regform.start_dt = now_utc()
         logger.info("Registrations for %s opened by %s", self.regform, session.user)
         flash(_("Registrations for {} are now open").format(self.regform.title), 'success')
+        new_dts = (self.regform.start_dt, self.regform.end_dt)
+        if new_dts != old_dts:
+            if not old_dts[1]:
+                log_text = 'Registration form "{}" was opened'.format(self.regform.title)
+            else:
+                log_text = 'Registration form "{}" was reopened'.format(self.regform.title)
+            self.event.log(EventLogRealm.event, EventLogKind.change, 'Event', log_text, session.user)
         return redirect(url_for('.manage_regform', self.regform))
 
 
 class RHRegistrationFormClose(RHManageRegFormBase):
     """Closes registrations for a registration form"""
 
     def _process(self):
+        old_dts = (self.regform.start_dt, self.regform.end_dt)
         self.regform.end_dt = now_utc()
         if not self.regform.has_started:
             self.regform.start_dt = self.regform.end_dt
         flash(_("Registrations for {} are now closed").format(self.regform.title), 'success')
         logger.info("Registrations for %s closed by %s", self.regform, session.user)
+        new_dts = (self.regform.start_dt, self.regform.end_dt)
+        if new_dts != old_dts:
```

Comment:
In which case would they be the same? I think in this RH you can unconditionally log closing the regform

Reply:
Right! Unconditionally logging now.